### PR TITLE
ROC-2062: Run integration tests against real database

### PR DESCRIPTION
**Pull request by geek-soft**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-2062

### Change description ###

This PR introduces real DB to run integration tests against.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/BaseTest.java [hunk 1]
```diff
@@ -38,6 +45,22 @@
     protected static final String REFERENCE_NUMBER = "000MC001";
     protected final Claim claimAfterSavingWithResponse = SampleClaim.getWithDefaultResponse();
 
+    @ClassRule
+    public static PostgreSQLContainer postgres = new PostgreSQLContainer("postgres:9.6")
+        .withDatabaseName("claimstore");
+
+    public static class Initializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
```
Comment (timja-kainos) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/BaseTest.java]: does this have to be public? can it just be static?

Comment (geek-soft) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/BaseTest.java]: I tried all other modifiers and only public seems to work. Private does not compile, remaining 2 fail at runtime.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/GetClaimByReferenceNumberTest.java [hunk 2]
```diff
@@ -2,65 +2,41 @@
...
 
-    @Test
-    public void shouldReturn500HttpStatusWhenInternalErrorOccurs() throws Exception {
-        given(claimRepository.getByClaimReferenceNumber(eq(REFERENCE_NUMBER))).willThrow(new RuntimeException("error"));
-
-        webClient
-            .perform(get("/testing-support/claims/" + REFERENCE_NUMBER))
-            .andExpect(status().isInternalServerError())
-            .andReturn();
+    private ResultActions makeRequest(String referenceNumber) throws Exception {
```
Comment (yarpo) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/GetClaimByReferenceNumberTest.java]: why don't we extract this method to some utils?

Comment (geek-soft) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/GetClaimByReferenceNumberTest.java]: Each test has its own `makeRequest` method which differs slightly by hitting different API paths.

**Pull request merged**